runtime/storage/cache/basic.go

```diff
@@ -0,0 +1,563 @@
+	return s.basicKeyspace.Set(ctx, key, val)
+}
+
+// SetIfNotExists sets the value stored at key to val, but only if the key does not exist beforehand.
+// If the key already exists, it reports an error matching ErrExists.
+func (s *FloatKeyspace[K]) SetIfNotExists(ctx context.Context, key K, val float64) error {
+	return s.basicKeyspace.SetIfNotExists(ctx, key, val)
+}
+
+// Replace replaces the existing value stored at key to val.
+// If the key does not already exist, it reports an error matching Miss.
+func (s *FloatKeyspace[K]) Replace(ctx context.Context, key K, val float64) error {
+	return s.basicKeyspace.Replace(ctx, key, val)
+}
+
+// GetAndSet updates the value of key to val and returns the previously stored value.
+// If the key does not already exist, it sets it and returns 0, nil.
+func (s *FloatKeyspace[K]) GetAndSet(ctx context.Context, key K, val float64) (oldVal float64, err error) {
+	return s.basicKeyspace.GetAndSet(ctx, key, val)
+}
+
+// GetAndDelete deletes the key and returns the previously stored value.
+// If the key does not already exist, it does nothing and returns 0, nil.
+func (s *FloatKeyspace[K]) GetAndDelete(ctx context.Context, key K) (oldVal float64, err error) {
+	return s.basicKeyspace.GetAndDelete(ctx, key)
+}
+
+// Delete deletes the key.
+// If the key does not already exist it returns nil.
+func (s *FloatKeyspace[K]) Delete(ctx context.Context, key K) error {
+	return s.basicKeyspace.Delete(ctx, key)
+}
+
+// Incr increments the number stored in key by delta,
+// and returns the new value.
+//
+// If the key does not exist it is first created with a value of 0
+// before incrementing.
+//
+// Negative values can be used to decrease the value,
+// but typically you want to use the Decr method for that.
+//
+// See https://redis.io/commands/incrbyfloat/ for more information.
+func (s *FloatKeyspace[K]) Incr(ctx context.Context, key K, delta float64) (newVal float64, err error) {
+	const op = "incr"
+	k, err := s.key(key, op)
+	if err != nil {
+		return 0, err
+	}
+
+	res, err := do[K, float64, *redis.FloatCmd](s.client, ctx, k, func(c cmdable) *redis.FloatCmd {
+		return c.IncrByFloat(ctx, k, delta)
+	}).Result()
+	err = toErr(err, op, k)
+	return res, err
+}
+
+// Decr decrements the number stored in key by delta,
+// and returns the new value.
+//
+// If the key does not exist it is first created with a value of 0
+// before decrementing.
+//
+// Negative values can be used to increase the value,
+// but typically you want to use the Incr method for that.
+//
+// See https://redis.io/commands/incrbyfloat/ for more information.
+func (s *FloatKeyspace[K]) Decr(ctx context.Context, key K, delta float64) (newVal float64, err error) {
+	const op = "decr"
+	k, err := s.key(key, op)
+	if err != nil {
+		return 0, err
+	}
+
+	res, err := do[K, float64, *redis.FloatCmd](s.client, ctx, k, func(c cmdable) *redis.FloatCmd {
+		return c.IncrByFloat(ctx, k, -delta)
```

Comment:
doesn't this support decr?

Reply:
No, there is no dedicated `DecrByFloat` in Redis

Reply:
👍 curious that they've added support for int decr